src/pages/patientView/clinicalInformation/MrnaExprRankCache.ts

```diff
@@ -5,81 +5,78 @@ import internalClient from "../../../shared/api/cbioportalInternalClientInstance
 
 export default class MrnaExprRankCache extends LazyCache {
 
-    protected populateCache(sampleToEntrezGeneIds:{ [sampleId:string]:Set<number> }, mrnaRankGeneticProfileId:string|null):Promise<boolean> {
+    protected async populateCache(sampleToEntrezGeneIds:{ [sampleId:string]:Set<number> }, mrnaRankGeneticProfileId:string|null):Promise<boolean> {
         const cache = this._cache;
-        return new Promise(resolve => {
-            // See which we need to fetch, and set "pending" for those data
-            const toQuery:{ [sampleId:string]:number[]} = {};
-            _.forEach(sampleToEntrezGeneIds, (entrezGeneIds:number[], sampleId:string) => {
-                cache[sampleId] = cache[sampleId] || {};
-                for (const entrezGeneId of entrezGeneIds) {
-                    if (!cache[sampleId].hasOwnProperty(entrezGeneId)) {
-                        toQuery[sampleId] = toQuery[sampleId] || [];
-                        toQuery[sampleId].push(entrezGeneId);
-                        cache[sampleId][entrezGeneId] = { status:"pending" };
-                    }
+        // See which we need to fetch, and set "pending" for those data
+        const toQuery:{ [sampleId:string]:number[]} = {};
+        _.forEach(sampleToEntrezGeneIds, (entrezGeneIds:number[], sampleId:string) => {
```

Comment:
sorry, i keep commenting on these iterators.  probably the array method forEach would stand the best chance of getting TS to infer the type.